### PR TITLE
feat: add torus integration

### PR DESCRIPTION
**Pull request by shekhar-shubhendu**

### CODE REVIEW

**Review by shmodarresi — DISMISSED** [review]

File: src/api/utils.ts [hunk 1]
```diff
@@ -102,7 +102,7 @@ export const toMarket = async (
     const yesPreference =
       Number(marketData.auctionRunningYesPreference ?? 1) /
       Number(marketData.auctionRunningQuantity ?? 1);
-    yesPrice = divideDown(yesPreference);
+    yesPrice = roundToTwo(divideDown(yesPreference));
```
Comment (tlebon) [inline src/api/utils.ts:105]: theres a function that combines these, used one line below

File: src/api/utils.ts [hunk 2]
```diff
@@ -209,22 +209,25 @@ export const normalizeGraphBetSingleOriginator = ({
...
+    return Object.keys(groupedBets).reduce((prev, marketId) => {
+      const lqtNode = orderByTxContext(groupedBets[marketId]);
+      const edges: BetEdge[] = R.pathOr([], [0, 'bets', 'betEdges'], lqtNode);
+      if (lqtNode.length > 0 && edges.length > 0) {
+        prev.push({
+          block: lqtNode[0].txContext.blockInfo.block,
+          quantity: Number(edges[0].bet.quantity),
+          marketId,
+          originator: address,
+          probability: roundToTwo(divideDown(Number(edges[0].bet.probability) * 100)),
```
Comment (tlebon) [inline src/api/utils.ts:224]: https://github.com/tzConnectBerlin/prediction-market-ui/pull/545/files#r719387830

File: src/api/utils.ts (hunk 1 above)
Comment (shekhar-shubhendu) [reply src/api/utils.ts:105]: I do not need `roundTwoAndTokenDown` here. it has different logic.

File: src/contracts/Market.ts [hunk 3]
```diff
@@ -382,7 +407,7 @@ export const basicRemoveLiquidity = async (
   slippage: number,
 ): Promise<string> => {
   const executionDeadLine = getExecutionDeadline();
-  const exitOp = await marketContract.methods.removeLiquidity(
+  const exitOp = marketContract.methods.removeLiquidity(
```
Comment (tlebon) [inline src/contracts/Market.ts:410]: why is await removed only from this specific method?

Comment (shekhar-shubhendu) [reply src/contracts/Market.ts:410]: 'cause we do not need to await there. we need to only await when we call `send`

File: .nvmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+16.10.0
```
Comment (tlebon) [inline .nvmrc:1]: is this necessary?

File: src/api/utils.ts (hunk 1 above)
Comment (tlebon) [reply src/api/utils.ts:105]: i misread

File: .nvmrc (hunk 4 above)
Comment (shekhar-shubhendu) [reply .nvmrc:1]: yes. for torus we need node > 14

**Comment by tlebon** [comment]
i havent tested but based off the code its fine, ill approve once i test interactions

**Comment by tlebon** [comment]
i guess this is fine, regular wallet interactions work fine. i cant test the torus wallet interactions without making an eve character 🤷🏻‍♂️

**Review by tlebon — APPROVED** [review]

**Pull request merged**